client/asset/dcr/dcr.go

```diff
@@ -2778,6 +2974,33 @@ func (dcr *ExchangeWallet) sendCoins(addr stdaddr.Address, coins asset.Coins, va
 	return tx, uint64(txOut.Value), err
 }
 
+// sendAll sends the maximum sendable amount (total input amount minus fees) to
+// the provided account as a single output, spending the specified coins.
+func (dcr *ExchangeWallet) sendAll(coins asset.Coins, destAcct string) (*wire.MsgTx, uint64, error) {
+	addr, err := dcr.wallet.InternalAddress(dcr.ctx, destAcct)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	baseTx := wire.NewMsgTx()
+	totalIn, err := dcr.addInputCoins(baseTx, coins)
+	if err != nil {
+		return nil, 0, err
+	}
+	payScriptVer, payScript := addr.PaymentScript()
+	txOut := newTxOut(int64(totalIn), payScriptVer, payScript)
+	baseTx.AddTxOut(txOut)
+
+	feeRate, err := dcr.feeRate(2)
+	if err != nil {
+		dcr.log.Debugf("sendAll feeRate error: %v, using fallbackFeeRate", err)
+		feeRate = dcr.fallbackFeeRate
```

Comment:
Probably could use `targetFeeRateWithFallback(2, 0)`.

We should probably also consider a latest fee rate cache internal to `ExchangeWallet` that can be passed in cases where we'd otherwise be using 0 for the fee suggestion

Reply:
I honestly wouldn't mind using an external API as a fallback.

https://explorer.dcrdata.org/insight/api/utils/estimatefee?nbBlocks=2

Reply:
Funny, I've been having the same thoughts recently, for all assets really.